### PR TITLE
Fix oil water case SOF2 + SWFN

### PR DESCRIPTION
**Pull request by totto82**
Fix for the first part of OPM/opm-simulators#3668

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

File: opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp [hunk 1]
```diff
@@ -927,17 +927,27 @@ class EclMaterialLawManager
...
-            for (size_t sampleIdx = 0; sampleIdx < sof3Table.numRows(); ++ sampleIdx)
-                SwSamples[sampleIdx] = 1 - sof3Table.get("SO", sampleIdx);
-
             effParams.setKrwSamples(SwColumn, normalizeKrValues_(tolcrit, swfnTable.getColumn("KRW")));
-            effParams.setKrnSamples(SwSamples, normalizeKrValues_(tolcrit, sof3Table.getColumn("KROW")));
             effParams.setPcnwSamples(SwColumn, swfnTable.getColumn("PCOW").vectorCopy());
+
+            if (!hasGas) {
+                const auto& sof2Table = tableManager.getSof2Tables().getTable<Sof2Table>(satRegionIdx);
+                // convert the saturations of the SOF3 keyword from oil to water saturations
```
Comment (bska) [inline opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp:936]: Tiny nit: This should probably refer to `SOF2` instead of `SOF3`.

**Review by bska — APPROVED** [review]
Looks good and clearly fixes an oversight.  Thanks a lot!  I just noticed a single inaccuracy in a comment.  Once that's fixed we'll merge this.

**Comment by totto82** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks.  I'll merge into master now.

**Pull request merged**